### ISSUE TITLE
Comp is not a function error

### ISSUE THREAD

**Issue by jan-vdm** [solidjs/solid-refresh#9]
Whenever I run the dev command it complains about Comp not being a function, however when I build and then serve the project it works perfectly fine with 0 console logs to indicate otherwise. At first I thought it is a component in my logged in state so I cleared my cache and forced a log out but it also happens on the log in screen. This is frustrating as I can still make changes, build a production version and serve it to check but would like to have hot reload working....
<img width="1724" alt="Screen Shot 2021-12-16 at 2 14 17 pm" src="https://user-images.githubusercontent.com/30859194/146302162-e8bc9bb4-2809-44bb-89ae-b1a8b98642be.png">

I checked the auth page and the container context but to be honest this is different error logs popping up vs the logged in state. And about a month ago it was working perfectly fine, just all of a sudden stopped...

Some context, I use contexts and use the children props within my context providers, this hasn't been an issue before but this is practically the only complex thing I've done with solidjs...

**Comment by dsimon-dev** [comment]
I had a similar issues and came here looking for solutions. I have a component inside a context object, and get the same error after HMR. While debugging I noticed that my `useContext` call returns an empty object, which should never be the case, and it only happens right after HMR. So in my case at actually has nothing to do with components specifically, accessing other context variables also throws a type error.

I found the following note inside the vite-plugin-solid repo readme:

> When you modify a file every state below this component will be reset to default state (including the **current file**). The state in parent component is preserved.

Since this states that the whole file is reset on HMR, I tried moving the `createContext` call to a separate file and import the context from there, and it worked! No more empty contexts.

I hope this is somewhat useful and related.

**Comment by LXSMNSYC** [comment]
Yes this issue may happen specially if the context is located in the same place as the components. I'll check if there's some decent, automatic workarounds for this.

**Comment by 53845714nF** [comment]
Hey, I tried Solidjs for the first time today and I get this error message too. What is the status of this issue?
I use `vite-plugin-solid` in version 2.7.0.
I also use `urql` could this possibly also cause the problem?

**Comment by ryansolid** [comment]
This issue was completed a while ago. If people still hit this issue, please submit a new issue as the cause of this error can be from a number of things.
